File: 0344-reverse-string/0344-reverse-string.java
class Solution {
    public void reverseString(char[] s) {
        int len = s.length;
        int start = 0;
        int end = len - 1;
        while(start < end){
            char temp = s[end];
            s[end] = s[start];
            s[start] = temp;

            start++;
            end--;
        }
    }
}

File: 0344-reverse-string/0344-reverse-string.cpp
class Solution {
public:
   void solve(vector<char>& s , vector<char>& rev ,int n){
        if(n==0) return;
        rev.push_back(s[n-1]);
        solve(s,rev,n-1);
   }
  
    void reverseString(vector<char>& s) {
        vector<char> rev;
        int n = s.size();
        solve(s,rev,n);

        for(int i=0;i<s.size();i++){
            s[i] = rev[i];
        }
    }
};